### ISSUE TITLE
Print a "giving up" message on deferred error

### ISSUE THREAD

**Issue by cdsmith** [google/codeworld#856]
When a program eventually crashes due to a deferred error, we want to print a message like:

    Giving up because of the error at program.hs:10:42.

We should be consistent, though.  Most programs, even with deferred errors, will crash in the first frame.  Essentially, failing in the first frame shouldn't look any different from a failed compile.

That means either we want to:

1. Also add a message like this when a build fails with non-deferred errors (something like `Giving up because I couldn't understand your code.`, since there's no single error location at fault in this case).  That way students get used to seeing the "giving up" message no matter the reason; OR
2. Only show the "Giving up..." message when the program runs for a while first, so that crashing on the first frame again doesn't look any different from failing to compile.

I guess, to implement option 2, one could grab a timestamp before starting the code, and then emit the "Giving up..." message if the crash is more than a couple seconds later.

**Comment by cdsmith** [comment]
I'm planning to implement this based on whether the canvas was shown.  Since the canvas is now only shown after the first frame is drawn, I think this will do what I want.
